pkg/middleware/middleware.go

```diff
@@ -182,6 +182,10 @@ func initContextWithBasicAuth(ctx *models.ReqContext, orgId int64) bool {
 }
 
 func initContextWithToken(authTokenService models.UserTokenService, ctx *models.ReqContext, orgID int64) bool {
+	if setting.LoginCookieName == "" {
```

Comment:
The problem at hand is that `ctx.GetCookie` will actually return **a** cookie when called with an empty string (`""`). I'll dig a big deeper tomorrow.